docs/source/using-discovery/slurm.md

```diff
@@ -268,24 +264,100 @@ srun --partition=gpu --nodes=1 --ntasks=1 --gres=gpu:1 --mem=1Gb --time=01:00:00
+The most basic configuration for a job array is as follows:
+:::{code}
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A.out
+#SBATCH --error=err/array_%A.err
+#SBATCH --array=1-6
+:::
+This command runs the same script six times using Slurm job arrays. Each job array has two additional environment variable sets. `SLURM_ARRAY_JOB_ID` (`%A`) is set to the first job ID of the array, and `SLURM_ARRAY_TASK_ID` (`%a`) is set to the job array index value.
+
+::::{tip}
+Generally, we want to pass the former as an argument for our script. If you are using R, you can retrieve the former using `task_id <- Sys.getenv("SLURM_ARRAY_TASK_ID")`. If you are using job arrays with Python, you can obtain the task ID using the following:
+
+:::{code} python
+import sys
+taskId = sys.getenv('SLURM_ARRAY_TASK_ID')
+:::
+::::
+
+In the previously defined Slurm header, the error and output file are overwritten whenever a "task" (one of the executions of the script through the job array) finishes. The modification below ensures that each "task id" will have its own output and error files.
+
+:::{code} bash
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A_%a.out
+#SBATCH --error=err/array_%A_%a.err
+#SBATCH --array=1-6
+:::
+When submitting an array with many dimensions, please use the "%" symbol to indicate how many tasks run simultaneously. For example, the following code specifies an array of 600 jobs, with 20 running at a time:
+
+:::{code} bash
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A_%a.out
+#SBATCH --error=err/array_%A_%a.err
+#SBATCH --array=1-600%20
+:::
+
+Whenever you specify the memory, number of nodes, number of CPUs, or other specifications, they will be applied to each task. Therefore, if we set the header of our submission file as follows:
+
+:::{code}
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A_%a.out
+#SBATCH --error=err/array_%A_%a.err
+#SBATCH --array=1-600%20
+#SBATCH --mem=128G
+#SBATCH --nodes=2
+:::
+Slurm will submit 20 jobs simultaneously. Each job, represented by a task ID, will use two nodes with 128GB of RAM each. In most cases, setting up a single task is sufficient.
+
+Lastly, we usually use job arrays for embarrassingly parallel jobs. If your case is such that the job executed at each job ID does not use any multi-threading libraries, you can use the following header to avoid wasting resources:
+
+:::{code}
+#!/bin/bash
+#SBATCH --partition=general
+#SBATCH --job-name=jarray-example
+#SBATCH --output=out/array_%A_%a.out
+#SBATCH --error=err/array_%A_%a.err
+#SBATCH --array=1-600%50  # 50 is the maximum number
+#SBATCH --ntasks=1
+#SBATCH --cpus-per-task=1
+#SBATCH --mem-per-cpu=2G ## debug prior to know how much RAM
+:::
+
+:::{warning}
+50 is the maximum number of jobs allowed to be run at once.
```

Comment:
for each user (researcher or account). Not sure on the language but should just be specific with this callout.

Reply:
Very nice! Thank you!